core/src/main/java/org/apache/iceberg/actions/BaseRewriteDataFilesAction.java

```diff
@@ -271,11 +273,13 @@ public RewriteDataFilesActionResult execute() {
   private void replaceDataFiles(Iterable<DataFile> deletedDataFiles, Iterable<DataFile> addedDataFiles,
                                 long startingSnapshotId) {
     try {
-      RewriteFiles rewriteFiles = table.newRewrite()
-          .validateFromSnapshot(startingSnapshotId)
-          .rewriteFiles(Sets.newHashSet(deletedDataFiles), Sets.newHashSet(addedDataFiles));
-      commit(rewriteFiles);
+      doReplace(deletedDataFiles, addedDataFiles, startingSnapshotId);
+    } catch (CommitStateUnknownException e) {
+      LOG.warn("Commit state unknown for files: {}, cannot clean up files because they may have been committed " +
+          "successfully.", Lists.newArrayList(addedDataFiles), e);
```

Comment:
Let's remove the list of data files from logs. That is going to be too long. Instead, let's use "Commit state unknown, cannot clean up files that may have been committed"